providers/Bandcamp/mod.ts

```diff
@@ -117,18 +128,18 @@ export default class BandcampProvider extends MetadataProvider {
 	}
 }
 
-export class BandcampReleaseLookup extends ReleaseLookup<BandcampProvider, AlbumPage> {
+export class BandcampReleaseLookup extends ReleaseLookup<BandcampProvider, ReleasePage> {
 	constructReleaseApiUrl(): URL | undefined {
 		return undefined;
 	}
 
-	async getRawRelease(): Promise<AlbumPage> {
+	async getRawRelease(): Promise<ReleasePage> {
 		if (this.lookup.method === 'gtin') {
 			throw new ProviderError(this.provider.name, 'GTIN lookups are not supported');
 		}
 
-		const webUrl = this.constructReleaseUrl(this.lookup.value);
-		const { content: release, timestamp } = await this.provider.extractEmbeddedJson<AlbumPage>(
+		const webUrl = this.constructReleaseUrl(this.id ?? this.lookup.value, this.lookup);
```

Comment:
```suggestion
		const webUrl = this.constructReleaseUrl(this.lookup.value, this.lookup);
```
I don't see why this should be necessary, both variables are guaranteed to contain the same value at this stage. Using the variable which doesn't require a non-null assertion (`this.id!`) would be my preference.
(Of course the situation will change if we ever remove the lookup method assertion above.)

Reply:
I guess I just wasn't sure if they were guaranteed to be the same. I changed it back to `this.lookup.value`, though that also means my change to move the `this.id` assignments is useless.

Reply:
I don't think it was useless, it should allow me to remove a few explicit ID assignments in other providers.